### PR TITLE
Dex file copy 1054

### PR DESCRIPTION
**Pull request by gulnaz-cdc**
testNG configuration for dex file copy and Meta data verify parameterization for all use cases

### CODE REVIEW

File: tests/integration/kotlin/src/test/kotlin/util/Metadata.kt [hunk 1]
```diff
@@ -1,16 +1,43 @@
...
+//                "filename" to file.name,
+//                "meta_destination_id" to Constants.TEST_DESTINATION,
+//                "meta_ext_event" to Constants.TEST_EVENT,
+//                "meta_ext_source" to "INTEGRATION-TEST"
+//            )
+//
+//        }
+
+        // this method is for reading the values from properties file and creating key value pair for metadata
+        fun generateRequiredMetadataForFile(file: File, propertiesFilePath: String): HashMap<String, String> {
```
Comment (cfarmer-fearless) [inline tests/integration/kotlin/src/test/kotlin/util/Metadata.kt:22]: The `file` parameter is unused here and can be removed

File: tests/integration/kotlin/src/test/resources/properties/daart-hl7.properties [hunk 2]
```diff
@@ -0,0 +1,9 @@
+filename=10KB-test-file
+meta_destination_id=daart
+meta_ext_event=hl7
+meta_ext_source=INTEGRATION-TEST
+original_filename=10KB-test-file
+#original_file_timestamp=timestampvalue
```
Comment (cfarmer-fearless) [inline tests/integration/kotlin/src/test/resources/properties/daart-hl7.properties:6]: Remove commented property

File: tests/integration/kotlin/src/test/resources/properties/dex-hl7-hl7ingress.properties [hunk 3]
```diff
@@ -0,0 +1,12 @@
+filename=10KB-test-file
+meta_destination_id=dex-hl7
+meta_ext_event=hl7ingress
+meta_ext_source=null
+meta_ext_sourceversion=null
+meta_username=test-automation
+original_filename=10KB-test-file
+original_file_timestamp=null
+message_type=CASE
+#message_type=ELR
```
Comment (cfarmer-fearless) [inline tests/integration/kotlin/src/test/resources/properties/dex-hl7-hl7ingress.properties:10]: Remove commented property

File: tests/integration/kotlin/src/test/resources/properties/ndlp-routineimmunization.properties [hunk 4]
```diff
@@ -0,0 +1,12 @@
+filename=10KB-test-file
+meta_destination_id=ndlp
+meta_ext_event=routineImmunization
+meta_ext_source=IZGW
+meta_ext_sourceversion=V2022-12-31
+#meta_ext_sourceversion=["V2022-12-31","V2023-09-01"]
```
Comment (cfarmer-fearless) [inline tests/integration/kotlin/src/test/resources/properties/ndlp-routineimmunization.properties:6]: Remove commented property

File: tests/integration/kotlin/src/test/resources/testNG.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd">
+<suite name="Parameterized Suite" parallel="methods" thread-count="1">
+    <test name="Generic Test Cases">
+        <parameter name="CONFIG_FILE" value="ndlp-rsvprevention.properties
+"/>
+        <classes>
+<!--            <class name="ProcStat">-->
```
Comment (cfarmer-fearless) [inline tests/integration/kotlin/src/test/resources/testNG.xml:7]: We should be able to keep all test classes commented in, and provide documentation on how to run all of them at once, and run only a specific one.

File: tests/integration/kotlin/src/test/kotlin/util/Utilities.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package util
+import org.json.JSONObject
+class Utilities{
+    fun extractMetadataFromResponseBody(jsonResponse: String): HashMap<String, String> {
```
Comment (cfarmer-fearless) [inline tests/integration/kotlin/src/test/kotlin/util/Utilities.kt:4]: This function can likely be removed if it's not needed after removing the proc stat code from `MetadataVerify.kt`.

**Review by cfarmer-fearless — CHANGES_REQUESTED** [review]

File: tests/integration/kotlin/src/test/resources/properties/dex-hl7-hl7ingress.properties (hunk 3 above)
Comment (gulnaz-cdc) [reply tests/integration/kotlin/src/test/resources/properties/dex-hl7-hl7ingress.properties:10]: there are two types of message_type CASE and ELR, we can do one at a time, this is for convenience. If a user wants to switch to the mrssage_type=ELR,  the user simply can comment out the first message type.

**Review by cfarmer-fearless — APPROVED** [review]

**Pull request merged**